### PR TITLE
Completed recipes page #23 

### PR DESCRIPTION
**Pull request by APM246**
## Change Summary
Completed recipes section of website, including:
- grid of recipes
- overview page for each recipe
- Slideshow modal
- All-steps page
- Download PDF button

 Closes a number of issues including #23, #54 and #197

## Other Information
Has been tested on all devices including mobile and tablets.

### CODE REVIEW

**Comment by CodeWithTheDoctor** [comment]
Not an urgent issue, but the "Let's cook" section has a visible scroll bar, 
I think removing it might be good, using the css down remove...

Also, it might be worth creating a custom, purple-coloured scrollbar... anyway, not super related to the issue and isn't really a big problem. I'm still in the process of reviewing the code.
```
/* Hide scrollbar for Chrome, Safari and Opera */
.no-scrollbar::-webkit-scrollbar {
    display: none;
}

/* Hide scrollbar for IE, Edge and Firefox */
.no-scrollbar {
    -ms-overflow-style: none;  /* IE and Edge */
    scrollbar-width: none;  /* Firefox */
}
```

**Comment by CodeWithTheDoctor** [comment]
image

I'm not sure if it is meant to look like this ahah... 😅

**Review by CodeWithTheDoctor — CHANGES_REQUESTED** [review]
The images also need to be in .webp format and need to be compressed using a tool such as https://squoosh.app/ as mentioned by Jeremiah on last Saturday.

File: components/Recipe/Overview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+  const adjustWidth = () => {
+    /* 
+    Find largest breakpoint under current screen size and send width to the Modal component. 
+    Default to largest screen size in the tailwind configs.
+
+    Might need a sort function in future but working fine for now
+      */
+
+    const fullConfig = resolveConfig(tailwindConfig)
```
[the remark this answers is not in the document]
Comment (innoc3ntt) [reply components/Recipe/Overview/index.tsx:82]: removed!

File: lib/types.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  imageGif: string
+  superPowers: string
+  foodGroup: string
+  location: string | string[]
+  facing?: string
+}
+
+type Breakpoints = 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl'
+
+//TODO: Fix the API to return ColorScheme as a required parameter. Just random values atm
```
[the remark this answers is not in the document]
Comment (innoc3ntt) [reply lib/types.ts:51]: been fixed

**Comment by APM246** [comment]
Not all recipes have their PDFs added to the repo (so download PDF button doesn't work for them)

**Review by chenga3 — DISMISSED** [review]

**Pull request merged**